File: original_php_project/src/User/Repository/UserAutologinTokenRepository.php
<?php

declare(strict_types=1);

namespace App\User\Repository;

use App\User\Entity\UserAutologinToken;
use Doctrine\ORM\EntityManagerInterface;
use Doctrine\Persistence\ObjectRepository;

final class UserAutologinTokenRepository
{
    private ObjectRepository $repository;

    public function __construct(EntityManagerInterface $em)
    {
        $this->repository = $em->getRepository(UserAutologinToken::class);
    }

    public function findByToken(string $token): ?UserAutologinToken
    {
        return $this->repository->findOneBy(['token' => $token]);
    }
}

File: target_java_project/src/main/java/com/goodhelp/user/domain/repository/UserAutologinTokenRepository.java
package com.goodhelp.user.domain.repository;

import com.goodhelp.user.domain.model.UserAutologinToken;

import java.util.Optional;

/**
 * Repository interface for UserAutologinToken entity.
 * Defines domain-level persistence operations.
 * 
 * <p>Implementation will be provided by the infrastructure layer
 * using Spring Data JPA.</p>
 */
public interface UserAutologinTokenRepository {

    /**
     * Find token by token string.
     */
    Optional<UserAutologinToken> findByToken(String token);

    /**
     * Find token by user ID.
     */
    Optional<UserAutologinToken> findByUserId(Long userId);

    /**
     * Save a token (create or update).
     */
    UserAutologinToken save(UserAutologinToken token);

    /**
     * Delete token by user ID.
     */
    void deleteByUserId(Long userId);

    /**
     * Delete a token.
     */
    void delete(UserAutologinToken token);

    /**
     * Check if token exists.
     */
    boolean existsByToken(String token);
}

